app/design/frontend/base/default/template/persistent/checkout/onepage/billing.phtml

```diff
@@ -45,14 +45,14 @@
                     <div class="field">
                         <label for="billing:company"><?php echo $this->__('Company') ?></label>
                         <div class="input-box">
-                            <input type="text" id="billing:company" name="billing[company]" value="<?php echo $this->escapeHtml($this->getAddress()->getCompany()) ?>" title="<?php echo Mage::helper('core')->quoteEscape($this->__('Company')) ?>" class="input-text <?php echo $this->helper('customer/address')->getAttributeValidationClass('company') ?>" />
+                            <input type="text" id="billing:company" name="billing[company]" value="<?php echo ($this->getAddress()->getCompany() ? $this->escapeHtml($this->getAddress()->getCompany()) : $this->getQuote()->getBillingAddress()->getCompany()); ?>" title="<?php echo Mage::helper('core')->quoteEscape($this->__('Company')) ?>" class="input-text <?php echo $this->helper('customer/address')->getAttributeValidationClass('company') ?>" />
```

Comment:
Is there a reason on why `$this->getQuote()->getBillingAddress()->getCompany()` is not echoed with `escapeHtml()`?

This also concerns all changes in this PR